experimental/LieAlgebras/src/simple_lie_algebra.jl

```diff
@@ -0,0 +1,174 @@
+#make root system string
+function string_root_system(L::SimpleLieAlgebra)
+  return string(root_system_type(L)[1]) * string(root_system_type(L)[2])
+end
+
+function Base.show(io::IO, ::MIME"text/plain", L::SimpleLieAlgebra)
+  io = pretty(io)
+  println(io, "Simple Lie algebra")
+  println(
+    io,
+    Indent(),
+    "of type $(string(root_system_type(L)[1])*string(root_system_type(L)[2]))",
+    Dedent(),
+  )
+  println(io, Indent(), "of dimension $(dim(L))", Dedent())
+  print(io, "over ")
+  print(io, Lowercase(), base_ring(L))
+end
+
+function Base.show(io::IO, L::SimpleLieAlgebra)
+  if get(io, :supercompact, false)
+    print(io, "Simple Lie algebra")
+    print(io, "$(string_root_system(L))")
+  else
+    io = pretty(io)
+    print(io, "Simple Lie algebra")
+    print(io, "$(string_root_system(L))", Dedent())
+    print(io, "over ", Lowercase())
+    print(IOContext(io, :supercompact => true), coefficient_ring(L))
+  end
+end
+
+###############################################################################
+#
+#   Arithmetic operations
+#
+###############################################################################
+
+# Binary operations
+
+function bracket(
+  x::SimpleLieAlgebraElem{C}, y::SimpleLieAlgebraElem{C}
+) where {C<:FieldElem}
+  check_parent(x, y)
+  L = parent(x)
+  mat = sum(
+    cxi * cyj * L.struct_consts[i, j] for (i, cxi) in enumerate(coefficients(x)),
+    (j, cyj) in enumerate(coefficients(y))
+  )
+  return L(mat)
+end
+###############################################################################
+#
+#   Constructors
+#
+###############################################################################
+
+@doc raw"""
+    lie_algebra(R::Field, S::Symbol, n::Int64, cached::Bool=true) -> SimpleLieAlgebra{elem_type(R)}
+
+Construct the simple Lie algebra over the field `R` with root system of type `S`
+The internally used basis of this Lie algebra is the Chevalley basis.
+"""
+function lie_algebra(R::Field, S::Symbol, n::Int64; cached::Bool=true)
+  return SimpleLieAlgebra{elem_type(R)}(R, S, n; cached)
+end
+
+###############################################################################
+#
+#   Chevalley basis
+#
+###############################################################################
+
+@doc raw"""
+    chevalley_basis(L::SimpleLieAlgebra{T}) -> Vector{Vector{SimpleLieAlgebraElem{T}}}
+
+Give the Chevalley basis of the simple Lie algebra `L` in three vectors, stating first the positive root vectors, 
+then the negative root vectors and finally the basis of the Cartan subalgebra. The order of root vectors corresponds
+to the order or the roots in the root system.
+"""
+function chevalley_basis(L::SimpleLieAlgebra)
+  RS = root_system(L)
+  n = length(RS.positive_roots)
+  B = basis(L)
+  #root vectors
+  r_plus = B[1:n]
+  r_minus = B[(n + 1):(2 * n)]
+  #basis for cartan algebra
+  h = B[(2 * n + 1):dim(L)]
+  return [r_plus, r_minus, h]
```

Comment:
```suggestion
    chevalley_basis(L::SimpleLieAlgebra{T}) -> NTuple{3,Vector{SimpleLieAlgebraElem{T}}}

Return the Chevalley basis of the simple Lie algebra `L` in three vectors, stating first the positive root vectors, 
then the negative root vectors, and finally the basis of the Cartan subalgebra. The order of root vectors corresponds
to the order of the roots in the root system.
"""
function chevalley_basis(L::SimpleLieAlgebra)
  RS = root_system(L)
  n = length(RS.positive_roots)
  B = basis(L)
  # root vectors
  r_plus = B[1:n]
  r_minus = B[(n + 1):(2 * n)]
  # basis for cartan algebra
  h = B[(2 * n + 1):dim(L)]
  return (r_plus, r_minus, h)
```
As the length of the return value is fixed, it is more efficient as a `NTuple` instead of a `Vector`. (+ typos fixed)